source/common/sagemath/library.sage

```diff
@@ -244,18 +244,17 @@ class TBIL:
         else:         
             xmin=0
             xmax=2*pi
-        if 'ticks' not in kwds.keys(): # or type(kwds['ticks']) is not sage.symbolic.expression.Expression:
+        if 'ticks' not in kwds.keys():
             delta=pi/4
             yticks=0.5
-        elif type(kwds['ticks']) is sage.symbolic.expression.Expression:
-            delta=kwds['ticks']
-            yticks=0.5
-        elif type(kwds['ticks']) is list and len(kwds['ticks'])==2:
-            delta=kwds['ticks'][0]
-            yticks=kwds['ticks'][1]
         else:
-            delta=pi/4
-            yticks=0.5
+            try:
+                iter(kwds['ticks']) # raises TypeError if not iterable
+                delta=SR(kwds['ticks'][0])
+                yticks=SR(kwds['ticks'][1])
```

Comment:
I don't think this is quite what we want, lots of things like dictionaries and strings are iterables.  I guess a string shouldn't error out since you can index it, but it won't behave as expected.  A dictionary will cause an error.  So I'm inclined to stick with the simpler improvement implemented in #671